src/Compilers/Core/Portable/Operations/ControlFlowGraphBuilder.cs

```diff
@@ -6500,6 +6500,41 @@ IOperation visitInvalidOperationExpression(IInvalidOperation invalidOperation)
             }
         }
 
+        public override IOperation VisitReDim(IReDimOperation operation, int? argument)
+        {
+            // We visit and rewrite the ReDim clauses in two phases:
+            //  1. Visit all the clause operands and indices and push them onto the eval stack.
+            //  2. Traverse the parts in reverse order, popping the indices and the operands from the eval stack for producing visited clauses.
+            EvalStackFrame frame = PushStackFrame();
+            foreach (var clause in operation.Clauses)
+            {
+                PushOperand(Visit(clause.Operand));
+                VisitAndPushArray(clause.Indices);
+            }
+
+            var clausesBuilder = ArrayBuilder<IReDimClauseOperation>.GetInstance(operation.Clauses.Length);
+            for (int i = operation.Clauses.Length - 1; i >= 0; i--)
+            {
+                IReDimClauseOperation clause = operation.Clauses[i];
+                var visitedIndices = PopArray(clause.Indices);
+                var visitedOperand = PopOperand();
+                var rewrittenClause = new ReDimClauseOperation(visitedOperand, visitedIndices, semanticModel: null,
+                    clause.Syntax, clause.Type, clause.ConstantValue, IsImplicit(clause));
+                clausesBuilder.Add(rewrittenClause);
+            }
+
+            clausesBuilder.ReverseContents();
+            PopStackFrame(frame);
+
+            return new ReDimOperation(clausesBuilder.ToImmutableAndFree(), operation.Preserve,
```

Comment:
>return new ReDimOperation(clausesBuilder.ToImmutableAndFree(), operation.Preserve, [](start = 12, length = 82)

I do not think this rewrite is going to give us correct order of evaluation in presence of control flow inside original constituent operations. For example, see the order of evaluation for this scenario:
```
Module Module1
    Sub Main()
        ReDim X(Get1())(Get2()), Y(Get3())
        System.Console.WriteLine("-------")
        ReDim Preserve X(Get1())(Get2()), Y(Get3())
    End Sub

    Property X(z As Integer) As Integer()
        Get
            System.Console.WriteLine("GetX")
            Return Nothing
        End Get
        Set(value As Integer())
            System.Console.WriteLine("SetX")
        End Set
    End Property

    Property Y As Integer()
        Get
            System.Console.WriteLine("GetY")
            Return Nothing
        End Get
        Set(value As Integer())
            System.Console.WriteLine("SetY")
        End Set
    End Property

    Function Get1() As Integer
        System.Console.WriteLine("Get1")
        Return 1
    End Function
    Function Get2() As Integer
        System.Console.WriteLine("Get2")
        Return 2
    End Function
    Function Get3() As Integer
        System.Console.WriteLine("Get3")
        Return 3
    End Function
End Module
```